demo/src/main/java/com/example/demo/contoller/StudentController.java

```diff
@@ -0,0 +1,99 @@
+
+package com.example.demo.contoller;
+
+import com.example.demo.dto.ResponseDTO;
+import com.example.demo.dto.StudentDTO;
+import com.example.demo.entity.StudentEntity;
+import com.example.demo.impl.IStudentDAO;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.dao.EmptyResultDataAccessException;
+import org.springframework.data.rest.webmvc.ResourceNotFoundException;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+@RestController
+@RequestMapping("/student")
+public class StudentController {
+	   
+	   private static String errorMessage = "No entity with id '{0}' found";
+	   
+	   @Autowired
+	   private IStudentDAO studentDAO;
+	   
+	   @PostMapping(value = "/1.0/save")
+	   @ResponseBody
+	   public ResponseEntity<ResponseDTO> save(@RequestBody StudentDTO studentDTO) {
+			 ResponseDTO responseDTO;
+			 HttpStatus status;
+			 try {
+				    studentDAO.saveStudentInfo(studentDTO);
+				    responseDTO = new ResponseDTO("Data Saved Successfully", HttpStatus.OK, HttpStatus.OK.value());
+				    status = HttpStatus.OK;
+			 } catch (Exception ex) {
+				    responseDTO = new ResponseDTO(ex.getMessage(), HttpStatus.NOT_FOUND, HttpStatus.NOT_FOUND.value());
+				    status = HttpStatus.NOT_FOUND;
+			 }
+			 return ResponseEntity.status(status).body(responseDTO);
+	   }
+	   
+	   @GetMapping(value = "/1.0/getstudentinfo")
+	   @ResponseBody
+	   public ResponseEntity<ResponseDTO> getStudentInfo(@RequestParam(name = "id") String id) {
+			 ResponseDTO responseDTO;
+			 HttpStatus status;
+			 try {
+				    StudentEntity studentEntity = studentDAO.getStudentInfo(id);
```

Comment:
How do we handle the Data not found situation here ?

Reply:
this is handle by "ResourceNotFoundException" exception